app/services/PaypalService.scala

```diff
@@ -88,34 +97,28 @@ class PaypalService(config: PaypalApiConfig, contributionData: ContributionData)
 
     val payment = new Payment().setIntent("sale").setPayer(payer).setTransactions(transactions).setRedirectUrls(redirectUrls)
 
-    Try {
-      val createdPayment = payment.create(apiContext)
-      createdPayment.getLinks.asScala
-    } match {
-      case Success(links) =>
-        val approvalLink = links.find(_.getRel.equalsIgnoreCase("approval_url")).map(l => Right(addUserActionParam(l.getHref)))
-        approvalLink.getOrElse(Left("No approval link returned from paypal"))
-      case Failure(exception) => Left(exception.getMessage)
+    val res = asyncExecute {
+      val links = payment.create(apiContext).getLinks.asScala
+      links.find(_.getRel.equalsIgnoreCase("approval_url")).map(l => addUserActionParam(l.getHref))
+    }.value map {
+      case Xor.Right(None) => Xor.left("No approval link returned from PayPal")
+      case Xor.Right(Some(link)) => Xor.right(link)
+      case Xor.Left(error) => Xor.left(error)
     }
+    XorT(res)
   }
 
   private def addUserActionParam(url:String) = Uri.parse(url).addParam("useraction", "commit").toString
 
-  def executePayment(paymentId: String, payerId: String): Either[String, Payment] = {
+  def executePayment(paymentId: String, payerId: String): XorT[Future, String, Payment] = {
     val payment = new Payment().setId(paymentId)
     val paymentExecution = new PaymentExecution().setPayerId(payerId)
-
-    Try(payment.execute(apiContext, paymentExecution)) match {
-      case Success(payment) =>
-        if (payment.getState.toUpperCase != "APPROVED") {
-          Left(s"payment returned with state: ${payment.getState}")
-        } else {
-          Right(payment)
-        }
-      case Failure(exception) =>
-        Logger.error("Unable to execute payment", exception)
-        Left(exception.getMessage)
+    val paymentResponse = asyncExecute(payment.execute(apiContext, paymentExecution)).value map {
```

Comment:
If you want to map over the `Xor`, according to the doc you should be able to use [transform](http://typelevel.org/cats/api/index.html#cats.data.XorT@transform[C,D](f:cats.data.Xor[A,B]=%3Ecats.data.Xor[C,D])(implicitF:cats.Functor[F]):cats.data.XorT[F,C,D]) here like that:
```
asyncExecute(payment.execute(apiContext, paymentExecution)).transform {
  case Xor.Right(blah) => some Xor result
}
```

and then there's no need for your final `XorT(paymentResponse)`